### PR TITLE
Room Assistant compatibility

### PR DESCRIPTION
**Pull request by demostenes**
* changed id field of ibeacon device reports to include minor number
* moved reporting of found devices to scanner callback - Room Assistant sends them right away without waiting for the scanner to finish; this also allows us to enable wantDuplicates flag to report a device multiple times during single scan, every time it's detected
* extracted more scanner parameters to settings file

### CODE REVIEW

File: ESP32-mqtt-room.ino [hunk 1]
```diff
@@ -305,10 +305,11 @@ class MyAdvertisedDeviceCallbacks: public BLEAdvertisedDeviceCallbacks {
 	void onResult(BLEAdvertisedDevice advertisedDevice) {
 
 		digitalWrite(LED_BUILTIN, LED_ON);
-		// Serial.printf("Advertised Device: %s \n", advertisedDevice.toString().c_str());
-		vTaskDelay(10 / portTICK_PERIOD_MS);
+		Serial.printf("Advertised Device: %s \n", advertisedDevice.toString().c_str());
+		if (mqttClient.connected()) {
+			reportDevice(advertisedDevice);
```
Comment (jptrsn) [inline ESP32-mqtt-room.ino]: Reporting device over MQTT during BLE scan will likely cause crashes. There is an issue with coexistence as both WiFi and Bluetooth use the same antenna. All devices are only reported after scan has ended for this reason.

Comment (demostenes) [reply ESP32-mqtt-room.ino]: I never got any crashes in my testing. Based on this http://bbs.esp32.com/viewtopic.php?f=13&t=6707&sid=30ede9a665f471d433dec8df587c24f0&start=30#p29419 I thought it should be ok if I keep scan window smaller that scan interval.

Comment (demostenes) [reply ESP32-mqtt-room.ino]: Would this patch be acceptable if I introduced a parameter in settings to select when devices will be reported?

Comment (jptrsn) [reply ESP32-mqtt-room.ino]: I have discounted the idea of reporting during the scan, as it caused no end of issues with stability. I can pull your branch and test locally. It's an insidious issue that only appears intermittently, and may be related to your local environment. For example, I'm in an apartment building where there are loads of wireless networks and bluetooth devices all comingled.

Comment (demostenes) [reply ESP32-mqtt-room.ino]: OK, I suppose you tested this more that me so I removed this change, leaving only modification of ibeacon id and scanner parameters in Setting.h. Is this acceptable? If yes - should I close this pull request, squash unnecessary commits and create new PR?

Comment (jptrsn) [reply ESP32-mqtt-room.ino]: Sorry for the delay in response - life has gotten in the way more than I had anticipated.

A new PR would certainly help me get these changes merged into the repo.

**Pull request closed without merging**